scala-package/infer/src/main/scala/ml/dmlc/mxnet/infer/ImageClassifier.scala

```diff
@@ -44,12 +45,19 @@ class ImageClassifier(modelPathPrefix: String,
 
   val classifier: Classifier = getClassifier(modelPathPrefix, inputDescriptors)
 
+  require(inputDescriptors.head.shape.length != 0,
+    "Please provide shape information in the descriptor")
+
+  require(!inputDescriptors.head.layout.isEmpty,
```

Comment:
no need for the I already changed the datadescriptor to have shape.length == layout.length